DFC.App.Banners.IntegrationTests/CustomWebApplicationFactory.cs

```diff
@@ -1,13 +1,52 @@
-﻿using Microsoft.AspNetCore.Hosting;
+﻿using DFC.App.Banners.Data.Models.ContentModels;
+using DFC.App.Banners.IntegrationTests.Extensions;
+using DFC.Compui.Cosmos.Contracts;
+using FakeItEasy;
+using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Mvc.Testing;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
 
 namespace DFC.App.Banners.IntegrationTests
 {
     public class CustomWebApplicationFactory<TStartup> : WebApplicationFactory<TStartup>
         where TStartup : class
     {
+        public CustomWebApplicationFactory()
+        {
+            this.MockCosmosRepo = A.Fake<ICosmosRepository<PageBannerContentItemModel>>();
+        }
+
+        internal ICosmosRepository<PageBannerContentItemModel> MockCosmosRepo { get; set; }
+
+        public HttpClient CreateClientWithWebHostBuilder()
+        {
```

Comment:
This is probably not needed, `builder.RegisterServices(this.MockCosmosRepo);` can move to `protected override void ConfigureWebHost(IWebHostBuilder builder)` method and you can keep using `var client = this.factory.CreateClient();` in your test classes.

Reply:
We can do that way as well. But current implementation gives more flexible and allows to use orginal statup (without mocks) with 'CreateClient' and startup with mocked classes with 'CreateClientWithWebHostBuilder'.

Reply:
but we'd only mock the external services, and those services will always be mocked in integration tests, so I don't see a need for that flexibility.

Reply:
that's fine. I am happy to update as to suggested.

Reply:
Updated code to resolve PR comments